include/net/coap.h

```diff
@@ -372,7 +372,7 @@ int coap_packet_parse(struct coap_packet *cpkt, uint8_t *data, uint16_t len,
  */
 int coap_packet_init(struct coap_packet *cpkt, uint8_t *data, uint16_t max_len,
 		     uint8_t ver, uint8_t type, uint8_t token_len,
-		     uint8_t *token, uint8_t code, uint16_t id);
+		     const uint8_t *token, uint8_t code, uint16_t id);
```

Comment:
The const change looks reasonable, could you add description about this change to release note as we are changing a public API.

Reply:
I'm not sure how to write release notes. Please check if the change I just pushed is what you expected.